documentation/develop/build-a-secure-extension.md

```diff
@@ -0,0 +1,107 @@
+---
+
+<!-- Page Hero Banner -->
+
+<section class="page-hero">
+<div class="module">
+<article class="module-content grid-x grid-padding-x">
+<div class="cell small-12">
+<div class="page-hero-description" markdown="1">
+<p class="section-title"><small>{{ page.topic }}</small></p>
+
+# Security best practices
+
+Here is a list of best practices you should follow to keep the users of your extension safe. If you don't follow these best practices your extension may fail the reviews on [addons.mozilla.org](https://addons.mozilla.org), preventing you from distributing your add-on or resulting in it being blocked from installation in Firefox.
+
+</div>
+<div class="page-hero-cta"></div>
+</div>
+</article>
+</div>
+</section>
+
+<!-- END: Page Hero Banner -->
+
+<!-- Single Column Body Module -->
+
+<section class="module">
+<article class="module-content grid-x grid-padding-x">
+<div class="cell small-12" markdown="1">
+
+- **Don’t inject or incorporate remote scripts**  
+  If you identify a service that you wish to use in your extension, don’t inject the service’s script from a remote source. Such an approach is risky, as the code could be changed without your knowing—and, importantly, without the knowledge and consent of the extensions user—compromising the security of your extension. You should therefore add a copy of the script into your extension’s code.
+- **Ensure you insert remote content safely**  
+  Make sure you follow best practices when your extension makes use of remote content:
+
+  - insert strings using safe native DOM manipulation methods: [document.createElement](/en-US/docs/Web/API/Document/createElement), [Element.setAtttribute](/en-US/docs/Web/API/Element/setAttribute), and [Node.textContent](/en-US/docs/Web/API/Node/textContent).
+  - use jQuery functions `attr()` and `text()` to insert strings.
+  - sanitize HTML content with  [DOMPurify](https://github.com/cure53/DOMPurify).
+  - use templating engine commands that escape any HTML before inserting it.
+
+  For more information, see [Safely inserting external content into a page](/en-US/Add-ons/WebExtensions/Safely_inserting_external_content_into_a_page).
+
+- **Use XHR for Google Analytics**  
+  If you want to add Google Analytics to your extension don't insert the Google Analytics JavaScript code. Rather, it’s recommended that the Google Analytics REST API is used in an XHR call, such as:
+
+  {% highlight javascript linenos %}
+  let request = new XMLHttpRequest();
+  let message =
+    "v=1&tid=" + GA_TRACKING_ID + "&cid= " + GA_CLIENT_ID + "&aip=1" +
+    "&ds=add-on&t=event&ec=AAA&ea=" + aType;
+
+  request.open("POST", "https://www.google-analytics.com/collect", true);
+  request.send(message);
+  {% endhighlight %}
+
+  You can find more information in the blog post [Using Google Analytics in Extensions](https://blog.mozilla.org/addons/2016/05/31/using-google-analytics-in-extensions/).
+
+- **Use the standard extension content security policy (CSP)**  
```

Comment:
Could a better choice of markup be used here. The bold text preceding the rest of the copy doesn't appear to provide much meaning. Maybe something like a `dl` would be a better choice, albeit you'd probably be needing to express that in pure HTML.

Alternatively could this be headings and paras? It could be nice to be able to link to a specific point.

Reply:
This is content ported from this page: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Security_best_practices
It will be up to the content team to make content edits once they have access to these new pages. For the purposes of this pull request we should leave content revisions for later review.